### PR TITLE
docs: moving docs to pages

### PR DESCRIPTION
**Pull request by npalm**
The documentation in our README have become quite extensive over the years. This PR is moving docs including examples and modules to a documentation site. The top-level readme is shrinked down to some basic / essential information.

- New top-level readme: https://github.com/philips-labs/terraform-aws-github-runner/tree/npalm/mkdocs?tab=readme-ov-file
- Documentation site: https://philips-labs.github.io/terraform-aws-github-runner

For examples and module documentation sources in example and module directories are linked.

### CODE REVIEW

**Comment by sdarwin** [comment]
the reorganization is looking very nice.  when possible it's probably better not to hard-code fulls URL such as "https://philips-labs.github.io/terraform-aws-github-runner/modules/public/ami-housekeeper/" if documentation is referring to itself it could use a relative path.

**Comment by sdarwin** [comment]
Minor spelling issue: Introduciton -> Introduction 

For each page of text if you quickly copy-paste it into Gmail, that has a built-in spell checker and will suggest what should be fixed.

**Comment by npalm** [comment]
> the reorganization is looking very nice. when possible it's probably better not to hard-code full URLs such as "https://philips-labs.github.io/terraform-aws-github-runner/modules/public/ami-housekeeper/" if documentation is referring to itself it could use a relative path.

would be much better, but it is hard to link from both the docs side and the readme's correctly.

File: .github/workflows/update-docs.yml [hunk 1]
```diff
@@ -32,9 +32,32 @@ jobs:
...
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+      - run: pip install mkdocs-material
+      - run: pip install mkdocs-material-extensions
+      - run: mkdocs gh-deploy --force -c -b npalm/test
```
Comment (koendelaat) [inline .github/workflows/update-docs.yml:63]: Replace npalm/test wit something more appropriate

File: mkdocs.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+site_name: GitHub Runners on AWS
+
+repo_name: philips-labs/terraform-aws-github-runner
+repo_url: https://github.com/philips-labs/terraform-aws-github-runner
+edit_uri: edit/npalm/mkdocs/docs/
```
Comment (koendelaat) [inline mkdocs.yaml:5]: replace `npalm` with something else

**Review by koendelaat — DISMISSED** [review]

File: mkdocs.yaml (hunk 2 above)
Comment (npalm) [reply mkdocs.yaml:5]: good catch, still pointing to the branch for testing.

**Review by koendelaat — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,461 +1,83 @@
...
-export interface GithubWorkflowEvent {
-  workflowJobEvent: WorkflowJobEvent;
-}
-```
-This extensible format allows more fields to be added if needed.
-You can configure the queue by setting properties to `workflow_job_events_queue_config`
+This [Terraform](https://www.terraform.io/) module creates the required infrastructure needed to host [GitHub Actions](https://github.com/features/actions) self-hosted, auto-scaling runners on [AWS spot instances](https://aws.amazon.com/ec2/spot/). It provides the required logic to handle the life cycle for scaling up and down using a set of AWS Lambda functions. Runners are scaled down to zero to avoid costs when no workflows are active.
 
-NOTE: By default, a runner AMI update requires a re-apply of this terraform config (the runner AMI ID is looked up by a terraform data source). To avoid this, you can use `ami_id_ssm_parameter_name` to have the scale-up lambda dynamically lookup the runner AMI ID from an SSM parameter at instance launch time. Said SSM parameter is managed outside of this module (e.g. by a runner AMI build workflow).
+insert image dark mode and light mode name is assets/runners.dark.png and assets/runners.light.png
```
Comment (Brend-Smits) [inline README.md:15]:
```suggestion

```

File: README.md [hunk 4]
```diff
@@ -1,461 +1,83 @@
...
-The module contains several submodules, you can use the module via the main module or assemble your own setup by initializing the submodules yourself.
+Cechkout the detailed istruction in the [Getting Started](https://philips-labs.github.io/terraform-aws-github-runner/getting-started/) section of the docs. On a high level, the following steps are required to get started:
+- Setup your AWS account
+- Create and configure a GitHub App
+- Download or build the required lambdas
+- Deploy the module using Terraform
+- Install the GitHub App to your organization or repositories and add your repositories to the runner group(s).
 
-The following submodules are the core of the module and are mandatory:
+Checkout the provide Terrafom examples in the [examples](./examples) directory for different scenarios.
```
Comment (Brend-Smits) [inline README.md:45]:
```suggestion
Checkout the provided Terraform examples in the [examples](./examples) directory for different scenarios.
```

File: README.md [hunk 5]
```diff
@@ -1,461 +1,83 @@
...
 
-- _[runner-binaries-syncer](./modules/runner-binaries-syncer/README.md)_ - Syncs the action runner distribution.
-- _[runners](./modules/runners/README.md)_ - Scales the action runners up and down
-- _[webhook](./modules/webhook/README.md)_ - Handles GitHub webhooks
-- _[multi-runner](./modules/multi-runner/README.md)_ - Creates multiple runner configurations in a single deployment
+## Configuration
 
-The following sub modules are optional and are provided as examples or utilities:
+Please check the [configuration](https://philips-labs.github.io/terraform-aws-github-runner/configuration.html) section of the docs for major configuration options. See the Terraform module documentation for all available options.
```
Comment (Brend-Smits) [inline README.md:49]: This link does not work for me. Should it?

**Review by Brend-Smits — COMMENTED** [review]
Minor changes, this is a welcoming change.

File: README.md (hunk 3 above)
Comment (npalm) [reply README.md:15]: lol prompt engineering :)

**Comment by npalm** [comment]
Pages branch set to gh-pages

**Review by Brend-Smits — APPROVED** [review]

**Review by koendelaat — APPROVED** [review]

**Pull request merged**